executor/executor_test.go

```diff
@@ -3360,3 +3360,13 @@ func (s *testSuite) TestDoSubquery(c *C) {
 	c.Assert(err, IsNil, Commentf("err %v", err))
 	c.Assert(r, IsNil, Commentf("result of Do not empty"))
 }
+
+func (s *testSuite) TestStrToDateBuiltin(c *C) {
```

Comment:
Please add some invalid cases, like:
`select str_to_date('69001010','%Y0%m0%d') from dual; `
`select  str_to_date('001001/10','%Y0%m/%d') from dual;`

Reply:
the warning message in TiDB is different from mysql. for example, the sql
`select str_to_date('00000/10/22','%Y/%m/%d') from dual;`

in mysql 
```
mysql> select str_to_date('00000/10/22','%Y/%m/%d') from dual;
+---------------------------------------+
| str_to_date('00000/10/22','%Y/%m/%d') |
+---------------------------------------+
| NULL                                  |
+---------------------------------------+
1 row in set, 2 warnings (0.00 sec)

mysql> show warnings;
+---------+------+------------------------------------------------------------------+
| Level   | Code | Message                                                          |
+---------+------+------------------------------------------------------------------+
| Warning | 1411 | Incorrect datetime value: '00000/10/22' for function str_to_date |
| Warning | 1411 | Incorrect datetime value: '00000/10/22' for function str_to_date |
+---------+------+------------------------------------------------------------------+
```
in TiDB
```
mysql root@localhost:(none)> select str_to_date('00000/10/22','%Y/%m/%d') from dual
+-----------------------------------------+
|   str_to_date('00000/10/22','%Y/%m/%d') |
|-----------------------------------------|
|                                  <null> |
+-----------------------------------------+
1 row in set
Time: 0.002s
mysql root@localhost:(none)> SHOW warnings;
+---------+--------+-------------------------------------------------+
| Level   |   Code | Message                                         |
|---------+--------+-------------------------------------------------|
| Warning |   1292 | Incorrect datetime value: '0000-00-00 00:00:00' |
+---------+--------+-------------------------------------------------+
```

should I just use the TiDB version to write some integration test cases ?

Reply:
@FateTHarlaown Sorry for this late reply, I did not notice this comment.
What I mentioned here means that we also need some integration tests for **other separators** except `/`.